python/python/tests/test_dataset.py

```diff
@@ -240,6 +240,21 @@ def test_take(tmp_path: Path):
     assert table2 == table1
 
 
+def test_take_duplicate_index(tmp_path: Path):
+    base_dir = tmp_path / "test"
+    arr1 = pa.array(["x", "y", "z"], pa.binary())
+    arr2 = pa.array(["y", "y"], pa.binary())
+
+    table1 = pa.Table.from_arrays([arr1], names=["b"])
+    lance.write_dataset(table1, base_dir)
+
+    dataset = lance.dataset(base_dir)
+    table2 = dataset.take([1, 1])
+
+    assert isinstance(table2, pa.Table)
+    assert table2 == pa.Table.from_arrays([arr2], names=["b"])
```

Comment:
Perhaps instead of hand writing the expected result, we can just test against PyArrow's `take`:

```suggestion
@pytest.mark.parametrize("indices", [[], [1, 1], [1, 1, 20, 20, 21], [21, 0, 21, 1, 0])
def test_take_duplicate_index(tmp_path: Path, indices: List[int]):
    table = pa.table({"x": range(24)})
    dataset = lance.write_dataset(table, tmp_path, max_rows_per_group=3, max_rows_per_fragment=9)
    expected = table.take(indices)
    assert dataset.take(indices) == expected
```

Also maybe nice to test more cases. There are different potential code paths depending on whether the indices span multiple groups / fragments. And always good to test the empty case (which we surprisingly don't have a case for yet!)